nixos/modules/system/boot/luksroot.nix

```diff
@@ -294,13 +294,11 @@ let
 
         new_response="$(ykchalresp -${toString yubikey.slot} -x $new_challenge 2>/dev/null)"
 
-        if [ ! -z "$k_user" ]; then
-            new_k_luks="$(echo -n $k_user | pbkdf2-sha512 ${toString yubikey.keyLength} $new_iterations $new_response | rbtohex)"
-        else
-            new_k_luks="$(echo | pbkdf2-sha512 ${toString yubikey.keyLength} $new_iterations $new_response | rbtohex)"
-        fi
-
-        echo -n "$new_k_luks" | hextorb > /crypt-ramfs/new_key
+        if [ -n "$k_user" ]; then
+	          echo -n "$k_user"
+	      else
+	          echo
+	      fi | pbkdf2-sha512 ${toString yubikey.keyLength} $new_iterations $new_response > /crypt-ramfs/new_key
```

Comment:
```suggestion
	          echo fi | pbkdf2-sha512 ${toString yubikey.keyLength} $new_iterations $new_response > /crypt-ramfs/new_key
```
?

Reply:
?

Reply:
Is it neccesary?

Reply:
I don't know. @SuperSandro2000 can you explain why you suggested this?

Reply:
He probably assumed that "fi  was being echoed but in reality we just want a blank line